client/src/js/services/grid/GridColumns.js

```diff
@@ -114,18 +114,19 @@ function GridColumnService(uiGridConstants, AppCache, Modal, util, $timeout) {
   Columns.prototype.hasEnoughColumns = function hasEnoughColumns(columns) {
     var grid = this.gridApi.grid;
     var visibleColumn = 0;
-    var defaultValueColumn = 3;
+    var defaultValueColumn = 1;
     var totalColumn = 0;
 
     angular.forEach(columns, function (visible, field) {
       var column = grid.getColumn(field);
-
+      
       /**
-        *This alternative structure checks if the first element of the grid is visible or not,
-        *finally to reinitialize the number of column by default
+        *This alternative structure checks if selectionRowHeaderCol or treeBaseRowHeaderCol are visible or not,
+        *finally to incremente the number of column by default
       */
-      if((column.name === "treeBaseRowHeaderCol") && !visible){
-        defaultValueColumn--;
+
+      if((column.name === "selectionRowHeaderCol" || column.name === "treeBaseRowHeaderCol") && visible){
```

Comment:
Do you have eslint installed? I think it asks for a space between the `if` and the `(`.  Like this:
```js
if (x) {
  // something
}
```

The reason is to try and separate function calls and control flow code.  It makes it easy for a programmer (without syntax highlighting) to realize at a glance that `if` is not a function call - it is a control flow statement.